infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/ILogReplicationVersionAdapter.java

```diff
@@ -26,4 +27,16 @@ public interface ILogReplicationVersionAdapter {
      * getNodeVersion() to determine if rolling upgrade has completed.
      */
     String getPinnedClusterVersion(TxnContext txnContext);
+
+    /**
+     * Get if the current deployment is SaaS or on-prem
+     * @return true if SaaS, otherwise false
+     */
+    boolean isSaasDeployment();
+
+    /**
+     * openVersionTable using the provided runtime
+     * @param runtime
+     */
+    void openVersionTable(CorfuRuntime runtime);
```

Comment:
why is this new method needed?

Reply:
Previously open table was done as part of the instance creation which took in runtime as a param. Now we do the same in an explicit call.
We now have a default constructor which can be used to query isSaas property in the plugin, and we later on open the version table to get the node version

Reply:
I see.. so this is the circular dependency.

runtime is required by this plugin not just to open this table but also the cluster version table in the real implementation.  So I suggest keeping it as a class variable for the adapter and using a setter to initialize it, instead of a separate call to open the tables.

Reply:
In the actual implementation, I saw that a ref to runtime was never saved. Instead we create a corfu store and store a ref to that as a class variable.
In addition to it, the only thing that is being moved out of the constructor of the adapter, is the opening of cluster version table. That's why I named the new api so.

If we have the setter for runtime, we'll need to include the logic of open table within "getPinnedClusterVersion". And since getPinnedClusterVersion is called in a for loop, we'll need to add some more logic to check if the table is already open.
So I am still inclined to keep it like this...but I'm open for discussion.

Reply:
ok, this seems simpler for now